lib/meilisearch/utils.rb

```diff
@@ -33,6 +33,21 @@ def self.parse_query(original_options, allowed_params = [])
       end
     end
 
-    private_class_method :parse
+    def self.message_builder(current_message, method_name)
+      "#{current_message}\nHint: It might not be working because maybe you're not up " \
+        "to date with the Meilisearch version that `#{method_name}` call requires."
+    end
+
+    def self.version_error_handler(method_name)
+      yield if block_given?
+    rescue MeiliSearch::ApiError => e
+      message = message_builder(e.http_message, method_name)
+
+      raise MeiliSearch::ApiError.new(e.http_code, message, e.http_body)
```

Comment:
Does this mean we are providing the hint even on `bad parameter` errors? And not only on route does not exists kind of errors?

Reply:
As I told you, the Ruby SDK has no such thing because the `CommunicationError` is only triggered when the SDK cannot connect with Meilisearch, eg. when Meilisearch is unreachable or the host is invalid.

To make it work, I would have to touch parts of the code I would not have time to do.

I agree it is better only to send in particular situations, but it is better than nothing 😅